referenceframe/frame.go

```diff
@@ -102,6 +102,9 @@ type Frame interface {
 	// Transform is the pose (rotation and translation) that goes FROM current frame TO parent's referenceframe.
 	Transform([]Input) (spatial.Pose, error)
 
+	// Interpolate interpolates the given amount between the two sets of inputs.
+	Interpolate(from, to []Input, by float64) ([]Input, error)
```

Comment:
major question: why is the output `[]Input` and not `[][]Input`? This seems like a mistake to me.
I assume we intend to replace 
```
interpolateSegment(ci *ik.Segment, resolution float64) ([][]frame.Input, error)
```
why is why I expect the outputs to match.

Reply:
Discussed in person